### PR TITLE
Add spellchecker

### PR DESCRIPTION
**Pull request by hannahapuan**
**Description of the change**: Adding spellchecker to .html and .md files!
**Reason for the change**: People misspell often in the docs! (Although it is great people contribute in general 😄 )

***Issue ref*** #3053 
This *DOES NOT* add to the build time (so I don't think it resolves the referenced issue, but it is a step in the right direction I hope!).

## Node Spellchecker

This is a spellchecker functionality using [node-markdown-spellcheck](https://www.npmjs.com/package/markdown-spellcheck) node package.

### Init
```
# (pull code updates before this)
# install updated npm package dependencies!
npm install
```

### Usage
```
# generates report of misspellings
npm run spellcheck

# generates report of misspellings of [filename]
npm run spellcheck-file [filename]
```
#### Example output:
```diff
     source/User_Guide/Marketing_Campaigns/lists.md
     38 | 1. From *Select Criteria* 
-drop-dwn
        choose from a list of availab 

    source/User_Guide/Settings/Subusers/settings.md
       28 | mber of times links from this 
-subuer’s
 emails were clicked this mont 
```
***NOTE***: this output is clearer in the command line.. I tried to make it understandable that the syntax coloring is red for incorrect words.

### Dictionary: English-US

See [http://wordlist.aspell.net/dicts/](http://wordlist.aspell.net/dicts/).
Also `.spelling` contains a list of custom words added to the dictionary.

### Contribution
This still needs work! Many, many words in tech jargon come back as incorrect.
The best way I've found to solve this issue without too much overhead work:
1. Run `npm run spellcheck`
2. Edit `.spelling` to include the words that are coming back incorrect but are correct!

@ksigler7

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/docs?pullRequest=3146) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/docs?pullRequest=3146) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/docs?pullRequest=3146) it.</sub>

**Comment by mbernier** [comment]
YESSSSSSS!!!!!

File: README.markdown [hunk 1]
```diff
@@ -172,3 +172,47 @@ payload in querystring format.
...
+```
+# (pull code updates before this)
+# install updated npm package dependencies!
+npm install
+```
+
+### Usage
+```
+# generates report of misspellings
+npm run spellcheck
```
Comment (mbernier) [inline README.markdown]: Can you please add 
`npm run spellcheck` to the "after_script" param in travis.yml?

This will cause it to run on every build ;)

**Comment by mamat-rahmat** [comment]
@mbernier Im not familiar with node. But, i guess the `markdown-spellcheck` dependency is not included in package.json. Maybe @hannahapuan installed globally on local and forget to add it?

**Comment by hannahapuan** [comment]
@mamat-rahmat the issue is that markdown-spellcheck returns status exit code 1 when there are errors. This was making the npm script fail -> the build failed. I fixed the `npm run spellcheck` script by appending `|| :` which makes it return exit code 0 on run. ¯\_(ツ)_/¯
build should be fine now.

**Comment by hannahapuan** [comment]
@mamat-rahmat ooh, I was mistaken, you are right
I looked at the error message and that's exactly what it is.
I did some testing in a different local directory and guess I forgot to install it in this package.json
Thanks for seeing that!! 👍

**Comment by hannahapuan** [comment]
Build is still failing, I think it's because it's finding too many issues in one file in the build.. Excluding checking some folders and seeing if the build will pass 😢

**Comment by hannahapuan** [comment]
Build is passing after I ignored the lintlink_log folder! 🎉  @mbernier ready to merge as far as I see :)

**Comment by ksigler7** [comment]
@hannahapuan this looks really good! the only thing I noticed was that running the script generates a file called `npm-debug.log` that we probably don't want to show up in our main tree, so I went ahead and added it to the `.gitignore` file. 

Thanks so much for doing all of this!!!

**Comment by mbernier** [comment]
@ksigler7 when you merge this, please watch the build to make sure that the Travis generation of this file isn't pushing this file to the server :(
If it is, we can add a command to travis.yml to remove the file before it pushes up the generated files.

**Comment by thinkingserious** [comment]
Hello @hannahapuan,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**